docs/source/examples/Image_reconstruction.rst

```diff
@@ -0,0 +1,320 @@
+    
+    # Train
+    model.fit_generator(train_dataloader, valid_dataloader, epochs=num_epochs, callbacks=callbacks, progress_options={'coloring': False})
+
+A random batch of the MNIST dataset images
+==================================
+
+Let's see some of the input samples, inside the training dataset.
+
+.. code-block:: python
+
+    inputs = next(iter(train_dataloader))[0]
+    input_grid = make_grid(inputs)
+    fig = plt.figure(figsize=(10, 10))
+    inp = input_grid.numpy().transpose((1, 2, 0))
+    plt.imshow(inp)
+    plt.show()
+
+.. image:: /_static/img/mnist_batch.png
+
+Reconstructed images after 3 epochs of training
+==================================
+
+In order to visually evaluate the quality of the results, here, we show the reconstruction result of the samples shown above
+
+.. code-block:: python
+
+    # Calculating predictions of the trained network on a batch
+    outputs = torch.tensor(model.predict_on_batch(inputs))
+    output_grid = make_grid(outputs)
+    fig = plt.figure(figsize=(10, 10))
+    out = output_grid.numpy().transpose((1, 2, 0))
+    plt.imshow(out)
+    plt.show()
+
+.. image:: /_static/img/mnist_3epoch.png
+
+Evaluation
+==================================
+
+One of the strong and useful tools in Poutyne is the evaluate class, which not only provides you with the evaluation metrics but also provides the ground truth and the prediction if the related arguments have been set to True (as below).
+
+.. code-block:: python
+
+    # evaluating thetrained network on test data
+    loss, predictions, ground_truth = model.evaluate_generator(test_dataloader, return_pred=True, return_ground_truth=True)
+
+In most computer vision applications, such as image reconstruction, it is very important to check the network's failures (or abilities, vice versa). The following part shows the input and the reconstructed images, as well as the reconstruction error map. Reconstruction error map shows which part of the image has not been reconstructed accurately.
+
+.. code-block:: python
+
+    sample_number = 2   # a sample from test dataset
+    
+    sample = ground_truth[sample_number][0]
+    sample_prediction_result_3epochs = predictions[sample_number][0]
+    
+    recunstruction_error_map_3epochs = sample - sample_prediction_result_3epochs  #reconstruction error map
+    fig, (ax1, ax2, ax3) = plt.subplots(1,3)
+    ax1.imshow(sample)
+    ax1.set_title('sample')
+    
+    ax2.imshow(sample_prediction_result_3epochs)
+    ax2.set_title('prediction')
+    
+    ax3.imshow(np.abs(recunstruction_error_map_3epochs))
+    ax3.set_title('reconstruction error')
+    plt.show()
+
+.. image:: /_static/img/rec_error_3epoch.png
+
+Resuming the training for more epochs
+==================================
+
+The code below shows how we can take advantage of Poutyne to resume training from a special epoch. For example, if we found the past epochs not enough, we can resume the training from the last done epoch, as below:
```

Comment:
Poutyne does not allow you to resume from any epoch. The `initial_epoch` argument just allows you to set the first epoch of the optimization at some epoch. Thus, I propose to reformulate this section as follow.

```suggestion
If we find the past epochs not enough, Poutyne allows you to resume the training from the last done epoch, as below:
```

Reply:
Also, since you used the `restore_best` argument in your ModelCheckpoint above, you need to load back the weights of the last epoch or else it will only continue your optimization from your best checkpoint. I think the best way would be to use the Experiment class instead of Model if you really want to show how to resume optimization with Poutyne.